src/services/notificationservice.cpp

```diff
@@ -67,17 +62,17 @@ bool NotificationService::removeNotification(unsigned int id)
         return false;
 }
 
-void NotificationService::setVibration(const QString v)
-{
-  if(v == "Strong")
-    m_vibrate = Strong;
-  else if( v == "Normal")
-    m_vibrate = Normal;
-  else
-    m_vibrate = None;
-}
-
-NotificationService::Vibrate NotificationService::getVibration()
+NotificationService::Vibrate NotificationService::strToVibrate(QString vibratename)
 {
-    return m_vibrate;
+    // anything not recognized will be interpreted as Strong
+    Vibrate retval = Strong;
+    
+    if (vibratename == vibrateSymbols[Ringtone].string) {
+       retval = Ringtone;
+    } else if (vibratename == vibrateSymbols[Normal].string) {
+        retval = Normal;
+    } else if (vibratename == vibrateSymbols[None].string) {
+        retval = None;
+    }
```

Comment:
Notice that if you were to add a VibrateMax element at the end of the enum (it's a fairly common pattern in C) you could implement this function like this:

```
for (Vibrate ret = Ringtone; ret < VibrateMax; ret ++) {
  if (VibrateStr[ret] == vibrateName)
    return ret;
}
return Strong;
```
The benfit of that approach is that this function no longer needs to change if the underlying enum and string array change, it would automatically adjust.
But that's really just a nitpick here,